crates/storage_models/src/events.rs

```diff
@@ -18,6 +18,27 @@ pub struct EventNew {
     pub primary_object_type: storage_enums::EventObjectType,
 }
 
+#[derive(Debug)]
+pub struct EventUpdate {
+    pub event_type: Option<storage_enums::EventType>,
+    pub event_class: Option<storage_enums::EventClass>,
+    pub is_webhook_notified: Option<bool>,
+    pub intent_reference_id: Option<String>,
+    pub primary_object_id: Option<String>,
+    pub primary_object_type: Option<storage_enums::EventObjectType>,
+}
```

Comment:
This should be an enum where the variants are the different types of updates that need to be done and the corresponding fields that need to be updated. refer to other tables.
```suggestion
pub enum EventUpdate {
    NotifyUpdate {
        is_webhook_notified: bool,
    }
}
```

Reply:
Other fields are never to be updated, we can remove them